pyvista/core/filters/data_set.py

```diff
@@ -4627,6 +4634,8 @@ def merge(
             grids = grid
             for grid in grids:
                 append_filter.AddInputData(grid)
+        elif grid is None:
+            append_filter.AddInputData(self)
```

Comment:
I assume this change is about this part of the docstring:

> Grid is updated in-place by default. Can be used to merge points of adjacent cells when no grids are input.

I would argue that the current code is better since this method also is described as `Join one or many other grids to this grid.`. No grid input would mean no change.  I'm also thinking about this use case:

```py
meshes = [ mesh for mesh in multi_mesh if some_condition(mesh)]
merged = grid.merge(meshes)
```

This would very surprising to find out that `grid` merged with itself if `meshes` is an empty list.

We can still merge a grid with itself in the current code.

Reply:
Looking closer, my example isn't true to since this PR only merges if `grid is None`.  I'm still concerned about the complexity of logic here and there might be some edge cases that will be unexpected.

Reply:
Good point. I will change the docstring without changing the code to convey this information.

Reply:
I'm not sure whether this behavior is intended for a good reason though, it may not be so obvious how to merge points within a dataset itself.  I mainly wanted to start a discussion on what is ideal since we have a change here.

Reply:
Can @pyvista/developers comment on this?

Reply:
There are two issues here:
1. `dataset.merge()` with no arguments seems non-intuitive.
2. We have no clear method for merging coincident points within a dataset.

We can resolve this in one of two ways:
1. Create a new method just for merging points (explicit behavior).
2. Modify `merge` as done in this PR (implicit) to allow for the merge of points.

I'm leaning towards (2) because we already specifically call out this use case in: `Can be used to merge points of adjacent cells when no grids are input.` If we add an example to the docstring merging points of a dataset, I think that should make it clear how to use the method in this manner.